tests/TestCase/Controller/ExportControllerTest.php

```diff
@@ -0,0 +1,564 @@
+        $response = $this->Export->export();
+        $content = $response->getBody()->__toString();
+        static::assertInstanceOf('Cake\Http\Response', $response);
+        static::assertEquals($expected, $content);
+    }
+
+    /**
+     * Data provider for `testCsvRows` test case.
+     *
+     * @return array
+     */
+    public function csvRowsProvider(): array
+    {
+        return [
+            'documents, all' => [
+                [
+                    'objectType' => 'documents',
+                ], // input
+                [
+                    'response' => [
+                        'data' => [
+                            0 => $this->testdata['input']['gustavo'],
+                            1 => $this->testdata['input']['johndoe'],
+                        ],
+                        'meta' => [
+                            'pagination' => [
+                                'page_items' => 2,
+                                'page_count' => 1,
+                            ],
+                        ],
+                    ],
+                    'data' => [
+                        0 => ['id', 'name', 'skills', 'category' => 'category'],
+                        1 => $this->testdata['expected']['gustavo'],
+                        2 => $this->testdata['expected']['johndoe'],
+                    ],
+                ], // expected
+            ],
+            'documents with ids' => [
+                [
+                    'objectType' => 'documents',
+                    'ids' => '999',
+                ], // input
+                [
+                    'response' => [
+                        'data' => [
+                            0 => $this->testdata['input']['gustavo'],
+                        ],
+                        'meta' => [
+                            'pagination' => [
+                                'page_items' => 1,
+                                'page_count' => 1,
+                            ],
+                        ],
+                    ],
+                    'data' => [
+                        0 => ['id', 'name', 'skills', 'category' => 'category'],
+                        1 => $this->testdata['expected']['gustavo'],
+                    ],
+                ], // expected
+            ],
+        ];
+    }
+
+    /**
+     * test 'csvRows'.
+     *
+     * @covers ::csvRows()
+     * @dataProvider csvRowsProvider()
+     *
+     * @param string|array $input The input for the function.
+     * @param string|array $expected The expected value.
+     *
+     * @return void
+     */
+    public function testCsvRows($input, $expected): void
+    {
+        // mock api getObjects.
+        $apiClient = $this->getMockBuilder(BEditaClient::class)
+            ->setConstructorArgs(['https://api.example.org'])
+            ->getMock();
+        $apiClient->method('getObjects')
+            ->willReturn($expected['response']);
+        ApiClientProvider::setApiClient($apiClient);
```

Comment:
is this necessary?
I suspect this causes other tests to fail, see https://travis-ci.org/bedita/manager/jobs/621166307 
Some mocked API client is then used in other test cases...

Reply:
resetting apiclient in tearDown... does the job